go/worker/registration/worker.go

```diff
@@ -319,6 +370,11 @@ func GetRegistrationSigner(logger *logging.Logger, dataDir string, identity *ide
 	return entity.ID, entitySigner, nil
 }
 
+// ScheduleShutdown requests that the node not register itself in the next epoch.
+func (w *Worker) ScheduleShutdown() {
```

Comment:
Should the notion of a "scheduled shutdown" be persisted to disk?

If i'm not wrong, the way it is done currently: if a node get's restarted while a shutdown was scheduled it would start normally and forget about the scheduled shutdown. I think it would make sense to handle this case.

Reply:
I believe this is done now.